File: csharp/Chapter09/Member.cs


namespace com.db4o.dg2db4o.chapter9
{
    class Member : Person
    {
        Account _account;
        
        public Member(string name, string phoneNumber, string email) : base(name,phoneNumber,email)
        {
            _account = new Account();
        }

        public Account Account
        {
            get
            {
                return _account;
            }
            set
            {
                _account = value;
            }
        }

        public override string ToString()
        {
            return base.Name + " (Member): " + System.String.Format("{0:C}", _account.Balance()); 
        }
    }
}

File: java/src/com/db4o/dg2db4o/chapter9/Member.java

package com.db4o.dg2db4o.chapter9;

import java.text.NumberFormat;

public class Member extends AbstractPerson {
    Account _account;
    
    public Member(String name, String phoneNumber, String email) {
        super(name,phoneNumber,email);
        _account = new Account();
    }
    
    public Account getAccount(){
        return _account;
    }
    
    public void setAccount(Account value) {
        _account = value;
    }
    
    public String toString() {
        return _name + " (Member): " + NumberFormat.getCurrencyInstance().format(_account.balance());
    }
}
